nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessGroupResource.java

```diff
@@ -2024,7 +2029,13 @@ public Response createProcessGroup(
                 }
             }
 
-            // Step 4: Resolve Bundle info
+            // Step 4: Replace parameter contexts if necessary
+            if (!keepExistingParameterContext) {
+                final ParameterContextReplacementUtil parameterContextReplacementUtil = ParameterContextReplacementUtil.getInstance(serviceFacade);
```

Comment:
This approach to creating an instance of ParameterContextReplacementUtil seems contrary to other conventions. It seems like it would be better to make the ParameterContext Replacement operations into an interface that could be injected. As it is, coupling the entire Service Facade to the Parameter Context Replacement instantiation is problematic.